framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -949,146 +951,175 @@ public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
     }
 
     /**
-     * Checks all the preconditions of the method invocation or variable access {@code tree} with
-     * element {@code invokedElement}.
+     * Checks all the preconditions of the method invocation {@code tree} with
+     * element {@code invokedMethodElement}.
+     *
+     * @param methodInvocationTree the MethodInvocationTree to check preconditions for.
+     * @param invokedMethodElement the ExecutableElement to retrieve the method preconditions from.
      */
-    protected void checkPreconditions(Tree tree,
-            Element invokedElement, boolean methodCall) {
-        checkPreconditions(tree, invokedElement, methodCall, null);
+    protected void checkMethodInvocationPreconditions(MethodInvocationTree methodInvocationTree,
+            ExecutableElement invokedMethodElement) {
+        if (invokedMethodElement == null) {
+            return;
+        }
+
+        checkPreconditions(methodInvocationTree, contractsUtils.getPreconditions(invokedMethodElement));
     }
 
     /**
-     * Checks all the preconditions of the method invocation or variable access {@code tree} with
-     * element {@code invokedElement}.
+     * Checks that all the given {@code preconditions} hold true immediately prior to
+     * the method invocation or variable access at {@code tree}.
```

Comment:
Conditions that must hold true immediately prior to the variable access. This is deliberately described in general terms, because it is up to the programmer writing a call to this method to determine what these conditions should be and what they mean. What this Javadoc is really saying is that this method can be called for a variable access, i.e. it will not enforce that `tree` is a `MethodInvocationTree`.
There could be postconditions for a variable write if a checker were to need them (the Lock Checker does not), but they would of course not be checked in this method.